### ISSUE TITLE
About the performance of the Attack on SimSwap

### ISSUE THREAD

**Issue by wkyhit** [shivangi-aneja/TAFIM#2]
Hi aneja, I trained the Face Protection model together with the SimSwap model under the configuration like this: 
`net_noise = 'unet_64' (default)
n_epochs = 100
perturb_wt=10 (default)
lr = 0.0001 (default)`
,due to the limitation of computing resources, I only used 2000 images from the CelebHQ dataset and trained the model for 100 epochs. The test results are shown below.
![26](https://user-images.githubusercontent.com/42362586/188312484-d052e35f-46ef-4f4e-a763-38d325bfcb53.png)
The second colum is the source image with pertubation and the noise seems quite obvious. I am wondering if this means I need to train for more epochs? 
And the testing script I used is here: https://github.com/wkyhit/TAFIM/blob/main/testing_scripts/test_protection_model_SimSwap.py

**Comment by shivangi-aneja** [comment]
Hi,
The Simswap model expects the input image in the range [0,1] and I think you might be un-normalizing the images incorrectly, which is why you see this white noise in the images. Maybe when calling `tensor2im` you use `fs=True` for SimSwap. The flag `fs` here stands for face-swapping. 

To control the amount of perturbation in the image, you can play around with the hyper-param `perturb_wt`. The higher its value, the less noise with be added to the input image but might degrade the target image quality.

**Comment by wkyhit** [comment]
Hi,
I set the **fs=True**, and the result is like this, the image colors now look good. But still, the original image has significant raster-like noise after the attack.
![43](https://user-images.githubusercontent.com/42362586/188315554-87275366-2a0a-4138-828a-c5191f6f3568.png)

I will try to set the **perturb_wt** param to little bit higher and see the result. Thanks for your kind reply.

**Comment by Lisszz** [comment]
Hi, I am facing the same raster-like noise in the "adv_noise" from the protection model but also in the "global_adv_noise" (global perturbation). I have run the training several times with different perturb_wt but it is the same problem for all. 

Do you have any idea what is causing this grid-like pattern? Thanks a lot in advance!

**Comment by shivangi-aneja** [comment]
Hi,

Sorry for the late reply. Using a higher weight for the parameter `pertrub_wt` in the loss term should fix the problem.
